lib/tree.rb

```diff
@@ -1,25 +1,41 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+class Tree
+  attr_reader :height, :age
+  MAXIMUM_AGE = 100
+  MAXIMUM_FRUIT_DIAMETER = 10
 
   def initialize
+    @height = 0.5
+    @age = 0
+    @apples = []
   end
 
   def age!
+    @age += 1
+    @height += 0.5 unless height >= 30
+    add_apples
   end
 
   def add_apples
+    if @age >= 3
+      rand(@age / 2...@age).times do
```

Comment:
Gotcha. I now understand it.

I just showed `rand(@age / 2...@age)` to another ruby dev at my desk a few minutes ago, and he misinterpreted it as integer/range, too. I don't have a specific suggestion yet. Just something to think about.